### DISCUSSION TITLE
Problem in mlps.run_rmse_test

### DISCUSSION THREAD

**Discussion by YuliangS** [MarsalekGroup/aml#28] [Q&A] [answered]
Hi
I am trying to run the example simulation on zundel. 01~04 are finished, but I meet a problem in "05-validation."
"mlps.run_rdf_test" and "mlps.run_vdos_test" work fine, but "mlps.run_rmse_test" does not work.
Here is the traceback:
![image](https://github.com/MarsalekGroup/aml/assets/105383439/64bd0668-3913-498c-b596-0672c9c4b891)

Could you help me solve this problems?

**Comment by beckobert** [comment, marked as answer]
`run_rdf_test` and `run_vdos_test` uses output from your CP2K MD simulation using the final C-NNP, `run_rmse_test` uses output from N2P2.
The `FileNotFoundError` would suggest that an output file from N2P2 is missing. Log and error files in the `predict-000` directory should be helpful in figuring out what went wrong.

**Reply by YuliangS** [reply]
Thank you for your reply.
Could you tell me what output files are used for "run_rmse_test" ?
I did not find "predict-000" directory. Should it appear in the "03-create-final-model" directory?

**Reply by beckobert** [reply]
The directories `predict-*` should be in the directory, where you run the `score-all.py` script.
I tried to reproduce your problem by running the function with a faulty N2P2 and the directories were in the `05-validation` directory, as expected. These directories will only be deleted, if the prediction was successful.
There you should find the two files `nnp-dataset-stdout.err` and `nnp-dataset-stdout.log`.
Alternatively, you can try to run the N2P2 binary `nnp-dataset` independently, to figure out the problem.

**Reply by YuliangS** [reply]
Sorry, could you tell me what the meaning of "physical unit" in N2P2?   I know it different for energy and force. 
Is it correct that " Energy in Hartree
                               Force in Hartree/Bohr". ?

**Reply by beckobert** [reply]
Please, see the official documentation of the N2P2 package: https://compphysvienna.github.io/n2p2/topics/units.html

> In principle the library and all applications are agnostic to a specific system of physical units. This means that numeric values in input files are processed unaltered. **Hence, it is the user’s responsibility to provide data in a consistent way.** For instance, the same length units must be used in training configurations (see [configuration file format](https://compphysvienna.github.io/n2p2/topics/cfg_file.html#cfg-file)) and in the [definition of symmetry functions](https://compphysvienna.github.io/n2p2/topics/descriptors.html#descriptors) (mind that some parameters, e.g. , need to be given in length units).

AML works internally in atomic units and all input files to N2P2 are written in atomic units.

**Reply by YuliangS** [reply]
I got it. Thank you so much.  
Sorry I have to bother you with another question. 
For the "aml.score" part, could you teach me how I can extract the raw data of RDF?  I want to plot the RDF by myself.

**Reply by cschran** [reply]
The rdfs are all saved as pickle files which you can load separately and plot to your liking.

If you load the pickle file, you get back the dictionary with all rdfs.

**Reply by YuliangS** [reply]
Thank you for the information!
But why does the RDF value approach 0 instead of approaching 1?

**Reply by cschran** [reply]
It is a gas phase molecule.
